wurst/_handles/Destructable.wurst

```diff
@@ -69,6 +69,81 @@ public function destructable.setOccluderHeight(real height)
 public function destructable.getName() returns string
 	return GetDestructableName(this)
 
+// Extensions
+
+tuple gate(destructable killable)
+/** Used to open a destructable gate */
+public function gate.open()
+	if (GetDestructableLife(this.killable) > 0)
+		KillDestructable(this.killable)
+
+/** Used to close a destructable gate */
+public function gate.close()
+	this.open()
+	DestructableRestoreLife(this.killable, GetDestructableMaxLife(this.killable), true)
+	SetDestructableAnimation(this.killable, "stand")
+
+/** Used to destroy a destructable gate */
+public function gate.kill()
+	this.open()
+	SetDestructableAnimation(this.killable, "death")
+
+/** Used to toggles a destructable gate open close state */
+public function gate.setOpen(bool flag)
+	if flag
+		this.open()
+	else
+		this.close()
+
+tuple elevator(destructable elevatable)
+constant ELEVATOR_HEIGHT_MIN = 1
+constant ELEVATOR_HEIGHT_MAX = 3		
+/** Used to move a destructable elevator one step up */
+public function elevator.up()
+	this.setHeight(this.getHeight()+1)
+
+/** Used to move a destructable elevator one step down */
+public function elevator.down()
+	this.setHeight(this.getHeight()-1)
+
+/** Used to move a destructable elevator to given height */
+public function elevator.elevatorSetHeight(int height)
+	this.setHeight(height)
+
+/** Used to get a destructable elevator height */
+public function elevator.getHeight() returns int
+	let height = ELEVATOR_HEIGHT_MIN + (this.elevatable.getOccluderHeight() / bj_CLIFFHEIGHT).toInt()
+	return height < ELEVATOR_HEIGHT_MIN or height > ELEVATOR_HEIGHT_MAX ? ELEVATOR_HEIGHT_MIN : height
+
+/** Returns elevator adjusted height based on given bounds */
+function isElevatorHeightInBounds(int height) returns bool
```

Comment:
unused private function? should be public? The one below too maybe?

Reply:
fixed